src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -73,15 +76,7 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    var oldTextVersion = oldDocument.GetTextVersionAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
-                    var newTextVersion = newDocument.GetTextVersionAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
-
-                    if (oldTextVersion != newTextVersion)
-                    {
-                        UpdateClassVersion();
-                        UpdateMembersVersion();
-                    }
-
+                    _documentChangedWorker.ContinueWith(_ => DocumentChangedAsync(oldDocument, newDocument), TaskScheduler.Default);
```

Comment:
Alternatively, you could just make this an async method and `await DocumentChangedAsync`.  Are we actually doing any computation as part of this that would occupy the UI thread, or are we just waiting on the results anyway?